src/ipelib/dynamo/module_weimer2005Ipe.f

```diff
@@ -0,0 +1,806 @@
+!-----------------------------------------------------------------------
+      subroutine read_potential2005Ipe(infile)
+!     
+! Read ascii data file W05scEpot.dat or W05scBpot.dat, written by 
+!   pro write_potential (write_data.pro)
+!
+      implicit none
+!
+! Args:
+      character(len=*),intent(in) :: infile
+!
+! Local:
+!
+      character(len=16) :: fname
+      integer :: i,lu=20
+      integer :: csize_rd,d1_rd,d2_rd
+!
+      open(lu,file=infile,status='old', ACCESS ='SEQUENTIAL')
+      read(lu,"(a)") fname
+      read(lu,"(28i3)") ab
+      read(lu,"(3i3)") csize_rd,d1_rd,d2_rd
+      if (csize_rd /= csize) then
+         write(6,"('>>> read_potential: file ',a                        &
+     &,': incompatable csize: ','csize_rd=',i4,' csize=',i4)")          &
+     &        fname,csize_rd,csize
+         stop 'csize'
+      endif
+      if (d1_rd /= d1_pot) then
+         write(6,"('>>> read_potential: file ',a,': incompatable d1: '  &
+     &,'d1_rd=',i4,' d1_pot=',i4)") fname,d1_rd,d1_pot
+         stop 'd1'
+      endif
+      if (d2_rd /= d2_pot) then
+         write(6,"('>>> read_potential: file ',a,': incompatable d2: '  &
+     &,'d2_rd=',i4,' d2_pot=',i4)") fname,d2_rd,d2_pot
+         stop 'd2'
+      endif
+      do i=1,csize
+         read(lu,"(6e20.9)") alschfits(:,i)
+      enddo
+      read(lu,"(2f10.3)") ex_pot
+      read(lu,"(28i3)") ls
+      read(lu,"(2i3)") maxl_pot,maxm_pot
+      read(lu,"(28i3)") ms
+      do i=1,csize
+         read(lu,"(6e20.9)") schfits(:,i)
+      enddo
+      close(lu)
+      end subroutine read_potential2005Ipe
+!-----------------------------------------------------------------------
+      subroutine schatable2005Ipe(infile)
+!
+! Read ascii data file SCHAtable.dat, written by pro write_scha
+!   (write_data.pro)
+!
+      implicit none
+!
+! Args:
+      character(len=*),intent(in) :: infile
+!
+! Local:
+!
+      character(len=16) :: fname
+      integer :: i,j,lu=20
+!
+      open(lu,file=infile,status='old', ACCESS = 'SEQUENTIAL')
+      read(lu,"(a)") fname
+      read(lu,"(2i3)") maxk_scha,maxm_scha
+      do i=1,d3_scha
+         do j=1,d2_scha
+            read(lu,"(6e20.9)") allnkm(:,j,i)
+         enddo
+      enddo
+      read(lu,"(8f10.4)") th0s
+      close(lu)
+      end subroutine schatable2005Ipe
+!-----------------------------------------------------------------------
+      subroutine read_bndy2005Ipe(infile)
+!
+! Read ascii data file W05scBndy.dat, written by pro write_bndy
+!   (write_data.pro)
+!
+      implicit none
+!
+! Args:
+      character(len=*),intent(in) :: infile
+!
+! Local:
+!
+      character(len=16) :: fname
+      integer :: rd_na,rd_nb,lu=20
+!
+      open(lu,file=infile,status='old', ACCESS = 'SEQUENTIAL')
+      read(lu,"(a)") fname
+      read(lu,"(2i3)") rd_na,rd_nb
+      if (rd_na /= na) then
+         write(6,"('>>> read_potential: file ',a,': incompatable na: ', &
+     &'rd_na=',i4,' na=',i4)") fname,rd_na,na
+         stop 'na'
```

Comment:
`STOP` statements _cannot_ be used in a parallel environment. Please modify the subroutine to return an error code if an error condition occurs, such us: missing file, I/O issue, and others.

Reply:
Hi Raffeale,  do you recommend that when we wrongly open the files, we only need to return an error code and write the error message, without any 'stop' or 'exit'?

Reply:
@ZhuxiaoLi66: yes, we should return an error code immediately after the error occurs. This error code must be propagated up to the IPE cap or driver, so that it can properly terminate the application.

Reply:
The error codes need to be brought out. We will address the problem altogether when Zhuxiao merges in her weimer2005_empirical branch. I have commanded out all the "stop" for now.